### ISSUE TITLE
LOBs are fully materialised

### ISSUE THREAD

**Issue by marschall** [Microsoft/mssql-jdbc#16]
BLOBs, CLOBs and NCLOBs are fully materialised into a single byte[]/String. In the case of large content (imagine a file stored in a BLOB) this puts quite a lot of pressure on both the allocator and the heap. It also defeats the point of streaming. Consider the following use case:

```java
try (InputStream stream = resultSet.getBlob("column_name").getBinaryStream()) {
    // stream content
}
```

**Comment by ajlam** [comment]
@marschall - Any suggestions on how to redesign this?

**Comment by marschall** [comment]
As far as I can tell the existing `ServerResultSet#getBinaryStream` and `SQLServerResultSet#getCharacterStream` methods provide streaming. One could imagine building a LOB implementation around these or their underlying concepts.

**Comment by v-suhame** [comment]
Thanks @marschall! That was a good catch, we will redesign it in the future releases.

**Comment by v-afrafi** [comment]
@marschall opened #172 regarding this issue.

**Comment by v-afrafi** [comment]
closed by pr#172. 
Thank you @marschall for pointing out this issue.
